File: 2022-2023/sourcecode/lab/lab11/OkHttpLoggingInterceptorBasicsJava/app/src/main/java/pl/udu/uwr/pum/okhttplogginginterceptorbasicsjava/PlaceholderService.java
package pl.udu.uwr.pum.okhttplogginginterceptorbasicsjava;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.DELETE;
import retrofit2.http.PATCH;
import retrofit2.http.PUT;
import retrofit2.http.Path;

public interface PlaceholderService {

    @PUT("posts/{id}")
    Call<Post> putPost(@Path("id") int postId, @Body Post post);

    @PATCH("posts/{id}")
    Call<Post> patchPost(@Path("id") int postId, @Body Post post);

    @DELETE("posts/{id}")
    Call<Void> deletePost(@Path("id") int postId);
}

File: 2022-2023/sourcecode/lab/lab11/RetrofitPutPatchDeleteBasicsKotlin/app/src/main/java/pl/udu/uwr/pum/retrofitputpatchdeletebasicskotlin/PlaceholderService.kt
package pl.udu.uwr.pum.retrofiturlbasicskotlin

import pl.udu.uwr.pum.retrofitpostbasicskotlin.Post
import retrofit2.Call
import retrofit2.http.*

interface PlaceholderService {
    @PUT("posts/{id}")
    fun putPost(@Path("id") postId: Int, @Body post: Post): Call<Post>

    @PATCH("posts/{id}")
    fun patchPost(@Path("id") postId: Int, @Body post: Post): Call<Post>

    @DELETE("posts/{id}")
    fun deletePost(@Path("id") postId: Int): Call<Void>
}
